src/dispatch/status.py

```diff
@@ -83,37 +83,30 @@ def status_for_error(error: Exception) -> Status:
     handler = _find_handler(error, _ERROR_TYPES)
     if handler is not None:
         return handler(error)
-
     # If not, resort to standard error categorization.
     #
     # See https://docs.python.org/3/library/exceptions.html
     status = Status.PERMANENT_ERROR
-    try:
-        # Raise the exception and catch it so that the interpreter deals
-        # with exception groups and chaining for us.
-        raise error
-    except IncompatibleStateError:
```

Comment:
I changed this to use `isinstance` because rethrowing the exception caused the stack trace calling `status_for_error` to be included in the exception traceback.

Reply:
Is it possible to `match` instead of calling `isinstance` again on each branch?

Reply:
See [a75d5e5](https://github.com/stealthrocket/dispatch-sdk-python/pull/72/commits/a75d5e580bb0a5c46edfbcd69990f5af86e680b0)